test/connectFactoryObservable.test.tsx

```diff
@@ -223,6 +231,125 @@ describe("connectFactoryObservable", () => {
       )
     })
 
+    it("allows sync errors to be caught in error boundaries with suspense", () => {
+      const errStream = new Observable(observer =>
+        observer.error("controlled error"),
+      )
+      const [useError] = connectFactoryObservable((_: string) => errStream)
+
+      const ErrorComponent = () => {
+        const value = useError("foo")
+
+        return <>{value}</>
+      }
+
+      const errorCallback = jest.fn()
+      const { unmount } = render(
+        <TestErrorBoundary onError={errorCallback}>
+          <Suspense fallback={<div>Loading...</div>}>
+            <ErrorComponent />
+          </Suspense>
+        </TestErrorBoundary>,
+      )
+
+      expect(errorCallback).toHaveBeenCalledWith(
+        "controlled error",
+        expect.any(Object),
+      )
+      unmount()
+    })
+
+    it("allows async errors to be caught in error boundaries with suspense", async () => {
+      const errStream = new Observable(observer => {
+        setTimeout(() => {
+          observer.error("controlled error")
+        }, 100)
+      })
```

Comment:
If we use a subject instead, we could save some time when running these tests - We can try to see if that would work.